keps/sig-cluster-lifecycle/kubeadm-config-ha-support.md

```diff
@@ -0,0 +1,149 @@
+---
+kep-number: TBD
+title: Augment Kubeadm Config to Enable Upgrades of HA Clusters
+authors:
+  - "@mattkelly"
+owning-sig: sig-cluster-lifecycle
+reviewers:
+  - "@timothysc"
+approvers:
+  - TBD
+editor:
+  - "@mattkelly"
+creation-date: 2018-03-08
+last-updated: 2018-03-14
+status: provisional
+see-also:
+  - [KEP kubeadm join --master workflow](https://github.com/kubernetes/community/pull/1707) (in progress)
+  - [Upgrading kubeadm HA clusters from 1.9.x to 1.9.y](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm-upgrade-ha)
+---
+
+# Augment Kubeadm Config to Enable Upgrades of HA Clusters
+
+## Table of Contents
+
+* [Augment Kubeadm Config to Enable Upgrades of HA Clusters](#augment-kubeadm-config-to-enable-upgrades-of-ha-clusters)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+     * [Goals](#goals)
+     * [Non-Goals](#non-goals)
+     * [Challenges and Open Questions](#challenges-and-open-questions)
+  * [Proposal](#proposal)
+     * [Implementation Details](#implementation-details)
+        * [Background](#background)
+        * [Adding Additional Master-Specific ConfigMaps](#adding-additional-master-specific-configmaps)
+           * [Key Design Considerations and Benefits](#key-design-considerations-and-benefits)
+              * [Parallel Node Creation](#parallel-node-creation)
+              * [Guaranteed Consistent kubeadm-config](#guaranteed-consistent-kubeadm-config)
+     * [Risks and Mitigations](#risks-and-mitigations)
+  * [Implementation History](#implementation-history)
+  * [Drawbacks](#drawbacks)
+  * [Alternatives](#alternatives)
+
+## Summary
+
+Currently, `kubeadm upgrade` of a master in a multi-master cluster will fail without workarounds because of a lack of node-specific master configuration in the `kubeadm-config` ConfigMap that is created at `init` time and later referenced during an upgrade.
+In particular, node-specific information is required in order for kubeadm to identify which control plane static pods belong to the current node during an upgrade.
```

Comment:
this sentence is a bit unclear to me. Could we better relate how "node-specific information" relate to a static pod?

Reply:
Updated